node/core/dispute-coordinator/src/scraping/mod.rs

```diff
@@ -195,7 +207,19 @@ impl ChainScraper {
 		{
 			Some(key_to_prune) => {
 				self.backed_candidates.remove_up_to_height(&key_to_prune);
-				self.included_candidates.remove_up_to_height(&key_to_prune);
+				let candidates_pruned = self.included_candidates.remove_up_to_height(&key_to_prune);
+				// Removing entries from inclusions per candidate referring to blocks before
+				// key_to_prune
+				for candidate_hash in candidates_pruned {
+					self.inclusions_per_candidate.entry(candidate_hash).and_modify(|inclusions| {
+						inclusions.retain(|inclusion| inclusion.0 >= key_to_prune);
+					});
+					if let Some(inclusions) = self.inclusions_per_candidate.get(&candidate_hash) {
```

Comment:
nit: Can you avoid searching twice in the HashSet for the same key? Maybe save the entry after the first get operation?